datafusion/sql/src/unparser/utils.rs

```diff
@@ -20,16 +20,23 @@ use datafusion_common::{
     tree_node::{Transformed, TreeNode},
     Result,
 };
-use datafusion_expr::{Aggregate, Expr, LogicalPlan};
+use datafusion_expr::{Aggregate, Expr, LogicalPlan, Window};
 
-/// Recursively searches children of [LogicalPlan] to find an Aggregate node if one exists
+/// One of the possible aggregation plans which can be found within a single select query.
+pub(crate) enum AggVariant<'a> {
```

Comment:
This assumes a SELECT query can exclusively have only a window function or an aggregate function but not both. A LogicalPlan can certainly have both, but I could not find an example of a single SELECT query without any nesting/derived table factors that is allowed to have both.